hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/AllowancesControllerIntegrationTest.java

```diff
@@ -263,12 +334,34 @@ void failTokenIdPresentWithoutAccount() {
     }
 
     @Test
-    void failWithUnsupportedMediaException() {
+    void nftAllowancesUnsupportedMediaException() {
         RestClient restClient = RestClient.create();
 
         // Performing the GET operation
         assertThrows(
                 HttpClientErrorException.MethodNotAllowed.class,
                 () -> restClient.post().uri(callUri, "0.0.1000").retrieve().body(NftAllowancesResponse.class));
     }
+
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                    "0.0.1000",
+                    "0.1000",
+                    "1000",
+                    "0.0x000000000000000000000000000000000186Fb1b",
+                    "0.0.0x000000000000000000000000000000000186Fb1b",
+                    "0x000000000000000000000000000000000186Fb1b",
+                    "0.0.AABBCC22",
+                    "0.AABBCC22",
+                    "AABBCC22"
+            })
+    void nftAllowancesNotFoundException(String accountId) {
+        RestClient restClient = RestClient.create();
+
+        // Performing the GET operation
+        assertThrows(
+                HttpClientErrorException.NotFound.class,
+                () -> restClient.get().uri(callUri, accountId).retrieve().body(NftAllowancesResponse.class));
+    }
```

Comment:
It's not sufficient to just check response code, we also need to verify error response body follows our expected format. Use assertj to verify additional exception details.

```suggestion
        assertThatThrownBy(() -> restClient.get()
                    .uri(uriParams, 1000)
                    .retrieve()
                    .body(NftAllowancesResponse.class)
                )
                .isInstanceOf(HttpClientErrorException.NotFound.class)
                .hasMessageContaining("some message")
                .asInstanceOf(InstanceOfAssertFactories.type(HttpClientErrorException.NotFound.class))
                .extracting(r -> r.getResponseBodyAs(Error.class).getStatus().getMessages().get(0))
                .returns("foo", ErrorStatusMessagesInner::getData)
                .returns("bar", ErrorStatusMessagesInner::getDetail)
                .returns("baz", ErrorStatusMessagesInner::getMessage);
```

We can put this in a helper method and use it to verify the other error cases.

Reply:
The error message extracted here is not the expected one.It is empty. I am working on getting this to work.

Reply:
It may not know how to resolve the type from the exception. Might have to register a `defaultStatusHandler()` on the `RestClientBuilder` with proper message converters.

Reply:
Currently only this method is returning an Error type in the service layer and not through the controller so I am checking for the detailed message only for the EntityNotFound exception.